### PR TITLE
Added update title interface

### LINKED ISSUE #368: Add functionality to update titles
## Description
Create view to update title when logged in.

## Acceptance criteria
- [ ] Investigate if backend supports updating title
- [ ] Create the view to update title

### PR DESCRIPTION
**Pull request by te4-adrian-almetun-smeds**
## ⚙️ What issue did you implement or fix?
Fixes #368

## 📜 Summary
- Added the ability to get a title based on it's ISBN number
- Added a interface for updating titles
- Added e2e tests

## 📝 Type of change

- [x] ✨ New feature or enhancement (non-breaking change which adds functionality)

## ✅ Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## 💩 How can this implementation be improved?
* Could use some unit tests

### CODE REVIEW

**Comment by te4-adrian-almetun-smeds** [comment]
The potential solution to code climates "Similar code", would most likely break backwards compatibility, if it where to be done in a way that actually isn't horrible. But rewriting it would also mean passing options to the function rendering it large and overly complicated instead.

**Review by te4-dennis-christensen — APPROVED** [review]
Cannot see anything that looks wrong to me, good job!

**Review by te4-jonathan-ronsten — APPROVED** [review]
Looks good to me. Can't see anything that looks bad. Nice!

File: frontend/cypress/integration/updateTitle.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    cy.get('div').contains('Kurslitteratur').click();
+    cy.get('[data-cy=subjectId]').click({force:true});
+    cy.get('div').contains('Programm').click();
+
+    cy.get('[data-cy=submit]').click();
+
+    cy.get('[data-jest=search]').type(bookTitle);
+    cy.get('[data-cy=formAction]').first().click();
+
+    // cy.get('[data-cy=reset]').click();
```
Comment (te4-adrian-almetun-smeds) [inline frontend/cypress/integration/updateTitle.js:30]:
```suggestion
```

File: frontend/src/views/AdminTitlesEditView.vue [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+</script>
+
+<style scoped>
+.flex {
+  display: flex;
+}
+
+.flex > div {
+  flex: 1;
+}
```
Comment (te4-adrian-almetun-smeds) [inline frontend/src/views/AdminTitlesEditView.vue:43]:
```suggestion
```

**Comment by itggot-linus-styren** [comment]
> The potential solution to code climates "Similar code", would most likely break backwards compatibility, if it where to be done in a way that actually isn't horrible. But rewriting it would also mean passing options to the function rendering it large and overly complicated instead.

This is a general issue, doesn't only affect this introduced change. Likely using a DI pattern or similar would be the best approach. I suggest to create a discussion regarding how to solve this. Maybe using https://github.com/inversify/InversifyJS could be the answer here, it would also simplify writing tests where it's necessary to mock all API calls.

**Pull request merged**